internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -313,11 +315,28 @@ public static Map<String, String> parseCommaSeparatedStringMapFromString(String
     try {
       Map<String, String> map = new HashMap<>();
       if (!value.isEmpty()) {
-        Arrays.stream(value.split(",")).map(s -> s.split("=")).forEach(strings -> map.put(strings[0], strings[1]));
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode rootNode = objectMapper.readTree(value);
+
+        rootNode.fieldNames().forEachRemaining(key -> {
+          JsonNode childNode = rootNode.get(key);
+          if (childNode.isObject()) {
+            map.put(key, childNode.toString());
+          } else {
+            map.put(key, childNode.asText());
+          }
+        });
       }
       return map;
-    } catch (Exception e) {
-      throw new VeniceException(fieldName + " must be key value pairs separated by comma, but value: " + value);
+    } catch (IOException jsonException) {
+      try {
+        Map<String, String> map = new HashMap<>();
+        Arrays.stream(value.split(",")).map(s -> s.split("=")).forEach(strings -> map.put(strings[0], strings[1]));
```

Comment:
Let's not carry this tech-debt over. Since it is admin-tool, it is okay to make breaking changes for now since it is only used ad-hoc

Reply:
Hey there, thanks for all the input I will make the changes you suggested. Can you explain a bit more about what you mean by tech-debt and how it applies? I have not heard this term before.

Thanks!

Reply:
Of course. Tech-debt is a term used to refer to anything that eventually becomes a maintenance overhead. In long-running software development, there are many iterations to a codebase and the behavior of the system evolves.

Similar to this PR, say you want to replace an existing functionality with another one. One way is to However, we've now accumulated users that are using the previous functionality of product and new features should not break the functionality that old users are using. So, we generally support both modes for some time till we can gracefully deprecate the old mode.

But in this PRs case, the code we're touching is the admin tool which is a tool we use to do ad-hoc maintenance operations against a running Venice cluster. The users of these tools are expected to be the operators of the Venice cluster and it is not widely used by users of Venice. So, in some cases like this, it is okay to break the backward compatibility in favor of reducing tech-debt.

Does this help? Also, I find Reddit ELI5's as a good resource for understanding any new concept. Here's the one for Tech-debt: https://www.reddit.com/r/explainlikeimfive/comments/5oxfe9/eli5_technical_debt/

Reply:
I appreciate the explanation, I just made some changes so let me know what you think and if anything needs to be improved still.

Reply:
Thanks. I left some more comments. But it is looking quite good now

Reply:
I realized the testParseMap test was covering this specific function, so I removed my redundant test and updated that test to check for the new functionality and added the other changes. Let me know if this is fine.

Reply:
Yeah, that works too. Thanks. The change mostly looks good. There are a few more failing tests. Can you please address those too?